python/cudf/cudf/core/column/numerical.py

```diff
@@ -350,12 +376,21 @@ def _can_return_nan(self, skipna: bool = None) -> bool:
 
     def _process_for_reduction(
         self, skipna: bool = None, min_count: int = 0
-    ) -> Union[ColumnBase, ScalarLike]:
+    ) -> Union[NumericalColumn, ScalarLike]:
         skipna = True if skipna is None else skipna
 
         if self._can_return_nan(skipna=skipna):
             return cudf.utils.dtypes._get_nan_for_dtype(self.dtype)
-        return super()._process_for_reduction(
+
+        super_obj = cast(
+            "cudf.core.column.ColumnBase",
+            (
+                super(NumericalColumn, self.nans_to_nulls())
+                if skipna
+                else super()
+            ),
+        )
+        return super_obj._process_for_reduction(
             skipna=skipna, min_count=min_count
         )
```

Comment:
I chose not to suggest this change last time around, but since @bdice started it :P
```suggestion
        col = self.nans_to_nulls() if skipna else self
        return super(NumericalColumn, col)._process_for_reduction(
            skipna=skipna, min_count=min_count
        )
```
This saves you some of the ugly mypy casting and having to save the output of super calls.